### ISSUE TITLE
The yum_cache plugin doesn't work with the epel-7-x86_64 config

### ISSUE THREAD

**Issue by RedDwarf69** [rpm-software-management/mock#274]
Fedora 30 with `mock-1.4.16-1.fc30.noarch` and `mock-core-configs-30.3-1.fc30.noarch`.

If I run both
```
mock -r epel-7-x86_64 --init
mock -r fedora-29-x86_64 --init
```

I get a nice dnf cache in `/var/cache/mock/fedora-29-x86_64/dnf_cache/`. But
```
$ ls -a /var/cache/mock/epel-7-x86_64/yum_cache/
.  ..  yumcache.lock
$
```
Interestingly epel-7-x86_64 seems to be using dnf
```
$ ls -l /var/lib/mock/epel-7-x86_64/root/var/cache/yum/
total 0
$ ls -l /var/lib/mock/epel-7-x86_64/root/var/cache/dnf/
total 56456
drwxr-sr-x. 4 root mock     4096 Jun 18 10:18 base-65e58d35cd0ad0ed
-rw-rw-r--. 1 root mock  7813461 Jun 18 10:18 base-filenames.solvx
-rw-rw-r--. 1 root mock  4067987 Jun 18 10:18 base.solv
drwxr-sr-x. 4 root mock     4096 Jun 18 10:18 epel-b7d9da81043aa7ab
-rw-rw-r--. 1 root mock 12253518 Jun 18 10:18 epel-filenames.solvx
-rw-rw-r--. 1 root mock     1542 Jun 18 10:18 epel-presto.solvx
-rw-rw-r--. 1 root mock  4820743 Jun 18 10:18 epel.solv
-rw-rw-r--. 1 root mock  4025458 Jun 18 10:18 epel-updateinfo.solvx
-rw-rw-r--. 1 root mock        2 Jun 18 10:19 expired_repos.json
drwxr-sr-x. 3 root mock     4096 Jun 18 10:18 extras-7bc8f4af39ce0c69
-rw-rw-r--. 1 root mock   222772 Jun 18 10:18 extras-filenames.solvx
-rw-rw-r--. 1 root mock    83801 Jun 18 10:18 extras-presto.solvx
-rw-rw-r--. 1 root mock   137274 Jun 18 10:18 extras.solv
drwxr-sr-x. 3 root mock     4096 Jun 18 10:18 sclo-7c9ee849a8fefabc
-rw-rw-r--. 1 root mock   877364 Jun 18 10:18 sclo-filenames.solvx
drwxr-sr-x. 3 root mock     4096 Jun 18 10:18 sclo-rh-3474da7af330b4d8
-rw-rw-r--. 1 root mock 12894399 Jun 18 10:18 sclo-rh-filenames.solvx
-rw-rw-r--. 1 root mock  2675227 Jun 18 10:18 sclo-rh.solv
-rw-rw-r--. 1 root mock   226977 Jun 18 10:18 sclo.solv
drwxr-sr-x. 4 root mock     4096 Jun 18 10:18 updates-39e09b98d9c64ce1
-rw-rw-r--. 1 root mock  5271727 Jun 18 10:18 updates-filenames.solvx
-rw-rw-r--. 1 root mock   882594 Jun 18 10:18 updates-presto.solvx
-rw-rw-r--. 1 root mock  1490714 Jun 18 10:18 updates.solv
$
```

Even if `package_manager` says `yum`
```
$ mock -r epel-7-x86_64 --debug-config 2>&1 | grep package_manager
config_opts['package_manager'] = 'yum'
                    'dir': '%(cache_topdir)s/%(root)s/%(package_manager)s_cache/',
                    'package_manager': 'yum',
                    'target_dir': '/var/cache/%(package_manager)s/'}}Finish: run
```

**Comment by RedDwarf69** [comment]
I'm guessing this is related to having `dnf-yum` installed in my Fedora 30 host system (the default).

FWIW I tried with `mock -r epel-7-x86_64 --init --bootstrap-chroot` and I got
```
ERROR: Command failed: 
 # /usr/bin/systemd-nspawn -q -M b4d4c227321042c49ac6ba061a5f4bbe -D /var/lib/mock/epel-7-x86_64-bootstrap/root -a --setenv=TERM=vt100 --setenv=SHELL=/bin/bash --setenv=HOME=/builddir --setenv=HOSTNAME=mock --setenv=PATH=/usr/bin:/bin:/usr/sbin:/sbin --setenv=PROMPT_COMMAND=printf "\033]0;<mock-chroot>\007" --setenv=PS1=<mock-chroot> \s-\v\$  --setenv=LANG=en_GB.UTF-8 --setenv=LC_MESSAGES=C.UTF-8 /usr/bin/yum --installroot /var/lib/mock/epel-7-x86_64/root/ --releasever 7 install @buildsys-build
There was a problem importing one of the Python modules
required to run yum. The error leading to this problem was:

   libelf.so.1: cannot open shared object file: No such file or directory

Please install a package which provides this module, or
verify that the module is installed correctly.

It's possible that the above module doesn't match the
current version of Python, which is:
2.7.5 (default, Apr  9 2019, 14:30:50) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]

If you cannot solve this problem yourself, please go to 
the yum faq at:
  http://yum.baseurl.org/wiki/Faq
```

**Comment by RedDwarf69** [comment]
FWIW it looks like the real "yum" RPM package will not be there in Fedora 31 any more (https://src.fedoraproject.org/rpms/yum/c/0d8aaac852e59cfa1177308132ac620e2d78eb5c)

And even if it does still exist, the real yum RPM package has been using `%{_bindir}/yum-deprecated` instead of `%{_bindir}/yum` since Fedora 22 (https://src.fedoraproject.org/rpms/yum/blob/f22/f/yum.spec).

**Comment by praiskup** [comment]
probably #268 as well

**Comment by praiskup** [comment]
Confirmed, mock should detect that /bin/yum is dnf;  and accordingly set the defaults for `config_opts['plugin_conf']['yum_cache_opts']['target_dir']`

**Comment by praiskup** [comment]
Or...., could `mock` just stop pretending it is using `yum` in such case (by using `class Yum`) but start using `dnf` right away?  At least if
- we are on fedora N >= ???
- /usr/bin/yum-deprecated doesn't exist

**Comment by RedDwarf69** [comment]
It may be a different issue, but when running `mock -r epel-7-x86_64 --init --bootstrap-chroot` with `yum` (`/usr/bin/yum-deprecated`) installed. I end up with this:

```
$ ls -l /var/cache/mock/epel-7-x86_64/yum_cache/*/packages/* | wc -l
104
$ ls -l /var/lib/mock/epel-7-x86_64/root/var/cache/yum/*/packages/* | wc -l
176
```

So the 104 packages downloaded for the bootstrap creation are correctly cached. But the ones downloaded for the final chroot creation aren't.

**Comment by praiskup** [comment]
We fixed this in #405, issue #392.  I entirely forgot about this one.
